### ISSUE TITLE
Provide copy-and-paste unparsed test "string"

### ISSUE THREAD

**Issue by Adal3n3** [codecov/engineering-team#2068]
## Problem to solve
Initially, we received feedback that our test names were too long and unreadable, so we parsed them into "Class Name" and "Test Name." However, we have since heard from developers who want to search for the failed test in their terminal, and the parsed names don't provide an easy copy-and-paste experience. Since we prioritize a developer-first experience, we should provide the full string that is easy to copy and paste.

For [example](https://github.com/codecov/codecov-api/pull/670#issuecomment-2220057495)
In our current PR:
![Screenshot 2024-07-12 at 5 10 58 PM](https://github.com/user-attachments/assets/baa77602-cfcb-45ca-8c89-f9448354a62c)
![Screenshot 2024-07-12 at 5 19 55 PM](https://github.com/user-attachments/assets/5c512999-e82c-41e4-9951-caa0db78479b)

`graphql_api/tests/test_owner.py::TestOwnerType::test_fetch_current_user_is_not_okta_authenticated
`
##### Steps to Reproduce / Current UX
1. update the PR comment design
2. tests dashboard

## Solution
Developers need the entire failed test string to allow them to copy and paste it into their terminal to find the test.

![Screenshot 2024-07-12 at 4 41 03 PM](https://github.com/user-attachments/assets/05a2531c-c030-402c-a9e6-bb5e63df09d6)
![Screenshot 2024-07-12 at 4 46 31 PM](https://github.com/user-attachments/assets/d77ae876-ae1f-47e6-9486-c58e4f5fdd54)

#### Action Checklist for Issue Creation
- [ ] Add the design project (to the right under "projects" > settings icon > select "design team")
- [ ] Share in relevant Slack channel (#discuss-codecov-product-design) for visiblity
- [ ] Add targeted launch date/delivery date as/if needed

**Comment by Adal3n3** [comment]
You can find the design here on [Figma](https://www.figma.com/design/TcppABudxnslgEmaXm0B9O/GH-2028?node-id=1-2)

![Image](https://github.com/user-attachments/assets/21cffb5a-0532-4bd8-b1c2-92ba07696fb8)

**Comment by Adal3n3** [comment]
This work might require 1-2 sprints.

**Comment by rohan-at-sentry** [comment]
This will require implementation on a language by language basis. For example

`Pytest` allows

1. Running all tests within a module `pytest test_mod.py` where `test_mod.py` is the whole module (contains many tests)
2. Running all tests in a directory `pytest testing/`
3. Running tests by keyword expressions `pytest -k "MyClass and not method`
4. Running a specific test within a module `pytest test_mod.py::TestClass::test_method`

for Pytest we'd likely want to support case 4

`Vitest` allows

1. Running all tests within a "file" `npm test -- SomeTestFileToRun`. 

There is no support for running individual test.

`Jest` allows

1. A single test to be run `node <path-to-jest> -i <your-test-file> -c <jest-config> -t "<test-block-name>"` {this is super hacky}
2. A single test `jest my-test`
3. Running tests that match a pattern `jest -t 'fix-order-test'`
4. Running all tests in a file `jest path/to/my-test.js`


Of the 4 options for jest, we'd want to support 3

`Ruby` allows

1. A single test to be run ` rails <path to test file.rb> -n "testname"`
2. All tests in a given file `rails <path to test file.rb>


TLDR

1. For python, we'll need test file path, class name, method name. (This is also true for Maven AFAICT)
2. For Ruby, Vitest and Jest, we'll need test file path and test name

**Comment by Adal3n3** [comment]
**For PHPunit test:**

This is the example from Michi: https://github.com/getsentry/sentry-symfony/actions/runs/10280796079/job/28454975345?pr=861

Michi will type his own script:
`vendor/bin/phpunit --filter End2EndTest::testMessengerCaptureHardFailure`

**Format:**
To rerun a specific failed test: `path/to/file/phpunit --filter ClassName::testMethodName`

**Comment by Adal3n3** [comment]
**For Jest and Vitest:**

- Jest and Vitest debugging workflows are identical.  
- Workflow 1: rerun the test suite ---> (eg. `test/sdk/api.test.ts` ) expensive to run but most commonly used
- Workflow 2: `test/**/*.test.ts` ---> will run everything
- Workflow 3: `test --watch` ---> run in watch mode (press F to only run failed tests)

![Image](https://github.com/user-attachments/assets/4c426174-74ca-43d2-909c-126f73033cee)

![Image](https://github.com/user-attachments/assets/ead76244-8d0e-46fd-a1ab-88d6543bbd9a)

![Image](https://github.com/user-attachments/assets/7b422ea6-5983-4d7c-af11-ca1779ffc640)

Meeting recording: https://sentry.rewatch.com/video/wnmln0deuhqkq5si-vitest-and-jest-c-p-test-name-september-26-2024

> All of these different packages have the same test name and the same path or the same file name because of the structure of our conventions. And so we actually use the full path to identify this properly for us.

We learned the test name is not unique so developers usually use the full file path to locate their test.

---

Format:
To rerun a specific failed test: `path/to/file > test name` (Confirmed w @joseph-sentry )

---

Note from @joseph-sentry 

> By default Jest does not include the filepath whereas vitest does

Joey has modified the doc so users will ask to include the filepath. 
![Image](https://github.com/user-attachments/assets/a6f257f0-5321-4a8a-bba8-623749780377)

We need to update the UI onboarding to include this change. @ajay-sentry here is the issue: https://github.com/codecov/engineering-team/issues/2605

**Comment by Adal3n3** [comment]
Completed w all languages.
